File: src/main/java/me/viniciusarnhold/altaria/events/StatsCounter.java
package me.viniciusarnhold.altaria.events;

import com.google.common.base.Stopwatch;

import java.util.ArrayList;
import java.util.Objects;
import java.util.concurrent.TimeUnit;
import java.util.concurrent.atomic.AtomicLong;

/**
 * Created by Vinicius.
 *
 * @since ${PROJECT_VERSION}
 */
public final class StatsCounter {

    private static final AtomicLong commandsHandled = new AtomicLong();

    private static final ArrayList<Long> commandHandleTime = new ArrayList<>();

    public static void addCommandHandled(long num) {
        commandsHandled.addAndGet(num);
    }

    public static long commandsHandled() {
        return commandsHandled.get();
    }

    public static void addCommandHandleTime(Stopwatch watch) {
        commandHandleTime.add(Objects.requireNonNull(watch).elapsed(TimeUnit.NANOSECONDS));
    }

    public static void addCommandHanndleTime(long nanos) {
        commandHandleTime.add(nanos);
    }

    public static long meanCommandHandleTime() {
        synchronized (commandHandleTime) {
            return (long) commandHandleTime.stream().mapToLong(l -> l).average().orElse(0L);
        }
    }

    public static long maxCommandHandleTime() {
        synchronized (commandHandleTime) {
            return commandHandleTime.stream().mapToLong(l -> l).max().orElse(0L);
        }
    }

    public static long minCommandHandleTime() {
        synchronized (commandHandleTime) {
            return commandHandleTime.stream().mapToLong(l -> l).min().orElse(0L);
        }
    }

}

File: core/src/main/kotlin/me/viniciusarnhold/altaria/events/StatsCounter.kt
package me.viniciusarnhold.altaria.events

import com.google.common.base.Stopwatch
import java.util.*
import java.util.concurrent.TimeUnit
import java.util.concurrent.atomic.AtomicLong

/**
 * Created by Vinicius.

 * @since ${PROJECT_VERSION}
 */
object StatsCounter {

    private val commandsHandled = AtomicLong()

    private val commandHandleTime = ArrayList<Long>()

    fun addCommandHandled(num: Long) {
        commandsHandled.addAndGet(num)
    }

    fun commandsHandled(): Long {
        return commandsHandled.get()
    }

    fun addCommandHandleTime(watch: Stopwatch) {
        commandHandleTime.add(Objects.requireNonNull(watch).elapsed(TimeUnit.NANOSECONDS))
    }

    fun addCommandHanndleTime(nanos: Long) {
        commandHandleTime.add(nanos)
    }

    fun meanCommandHandleTime(): Long {
        synchronized(commandHandleTime) {
            return commandHandleTime.stream().mapToLong { l -> l }.average().orElse(0.0).toLong()
        }
    }

    fun maxCommandHandleTime(): Long {
        synchronized(commandHandleTime) {
            return commandHandleTime.stream().mapToLong { l -> l }.max().orElse(0L)
        }
    }

    fun minCommandHandleTime(): Long {
        synchronized(commandHandleTime) {
            return commandHandleTime.stream().mapToLong { l -> l }.min().orElse(0L)
        }
    }

}
